### ISSUE TITLE
Missing errors on extra splat arguments to callables with paramspec args

### ISSUE THREAD

**Issue by samwgoldman** [microsoft/pyright#9615]
When calling a callable that uses a param spec, we must pass the appropriate args/kwargs. Passing additional args is not valid (ignoring Concatenate). Pyright allows extra arguments to slip through via splat arguments.

In the following two examples, we can pass splat args and splat kwargs having the wrong type without any error. My expectation is that Pyright should error at both the `*xs` args with some kind of message.

```
from typing import Callable

def f1[**P, R](g: Callable[P, R], xs: list[int], *args: P.args, **kwargs: P.kwargs) -> R:
    return g(*xs, *args, **kwargs)

def f2[**P, R](g: Callable[P, R], xs: dict[str,int], *args: P.args, **kwargs: P.kwargs) -> R:
    return g(*args, **xs, **kwargs)
```
[Playground](https://pyright-play.net/?strict=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoDCAhgDYlEBGJApgFC0Am1wUwA2gFQcAKANFACUAugAo0ALkKlyVamz6Ch-AB4BnSSSSqYbVDCVQOREGnVRuAOmOn%2BXANYB3a2cuPnASigBaAHyDxtFBBUCDUMACuIChQaCIcarbOthxuJqru9EwswACMnDz8wmKSxGSUNPKFBmqSDEgAxjraILx6BkZpkpZJhilOneYWqaaevv6BwaERUTFxPVwJvcPpQA)

Tested using the Playground only, using latest (1.1.391).

**Comment by erictraut** [comment]
Pyright is working as intended here, so I don't consider this a bug. The behavior is here is consistent with other situations where unpacking (splatting) is used with an argument of unknown length. Since the length could be zero, it would be a false positive to report this as an error.

```python
def f3(x: int):
    ...

def f4(*args: str):
    f3(1, *args) # No error

def f5(args: dict[str, str]):
    f3(1, *args) # No error

def f6(args: tuple[str]):
    f3(1, *args) # Error because tuple has known length
```

Note that the behavior in my code sample above is the same in both pyright and mypy.

There's a reasonable question about whether a type checker should be conservative (avoiding potential false positives) or not (avoiding potential false negatives) in this situation. I can make good arguments for both cases. If the typing community can converge on a consensus, I'd be happy to change pyright's behavior to match, but it should be done consistently.

I'd like to see this question answered and standardized as part of the typing spec. If you'd like to spearhead such a discussion and standardization effort, feel free to start a discussion in the [Python typing forum](https://discuss.python.org/c/typing/32).

Unless / until this behavior is standardized, I don't plan to make a change in pyright's behavior. Doing so would risk unnecessary churn for pyright users who rely on the current behavior. For that reason, I'm going to close this issue.

**Comment by samwgoldman** [comment]
Thanks, Eric. I agree it makes sense to discuss this through the lens of the typing spec instead. I won't have time to do it until after the holidays, but my thought is that `*` splat args with indeterminate length would consume all remaining positional and vararg params. If all positional params are already consumed, then a splat would be allowed, as in your examples, since it could be empty.

Mypy and Pyright already seem to follow the described behavior for concrete functions (note the splat arg comes first):

```
def f3(x: int):
    ...

def f6(*args: str):
    f3(*args, 1) # Error

def f7(args: dict[str, str]):
    f3(*args, 1) # Error
```

and Mypy seems to behave similarly for the paramspec case as well.
